Firebase.cpp

```diff
@@ -15,100 +15,158 @@
 //
 #include "Firebase.h"
 
-const char* firebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
-const uint16_t firebasePort = 443;
+namespace {
+const char* kFirebaseFingerprint = "7A 54 06 9B DC 7A 25 B3 86 8D 66 53 48 2C 0B 96 42 C7 B3 0A";
+const uint16_t kFirebasePort = 443;
+}  // namespace
 
-Firebase::Firebase(const String& host) : _host(host) {
-  _http.setReuse(true);
+Firebase::Firebase(const String& host) : connection_(host) {
 }
 
 Firebase& Firebase::auth(const String& auth) {
-  _auth = auth;
+  connection_.auth(auth);
   return *this;
 }
 
-String Firebase::get(const String& path) {
-  return sendRequest("GET", path);
+FirebaseResult Firebase::get(const String& path) {
```

Comment:
While these examples flow very well and look great the problem is this style assumes the calls succeed, which isn't great for remote services. I think the resulting use will be uglier but safer if we encourage checking for errors.

That said I agree whole heartedly with making reading/decoding optional, the return value for a lot of these calls seems unnecessary (like echoing back the value on a put)